THORN-1830: arquillian ignores swarm.http.port setting in project-defaults.yml

roguexz opened the pull request:
Motivation
----------
Thorntail aims at simplifying the whole development experience. The ability to test code using the prescribed (Arquillian based) methods, right from within the IDE plays a huge role in the experience. I noticed that when running Arquillian based test cases from within IntelliJ, the resources (project-defaults.yml, etc.) were not being picked up as part of the built archive. More details about the IDE behavior can be found in the issue.

Modifications
-------------
Updated the scenario generator to not filter out the resource folders.

Result
------
I can now run Arquillian based tests from within the IDE itself.

- [x] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [x] [v2] Have you created a [JIRA](https://issues.jboss.org/browse/THORN) and used it in the commit message?
- [ ] [v4] Have you created a [GitHub Issue](https://github.com/thorntail/thorntail/issues) and used it in the commit message?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/thorntail/thorntail/pulls) for the same issue?
- [x] Have you built the project locally prior to submission with `mvn clean install`?

-----

Commit:
THORN-1830: arquillian ignores swarm.http.port setting in project-defaults.yml

Motivation
----------
Thorntail aims at simplifying the whole development experience. The ability to test code using the prescribed (Arquillian based) methods, right from within the IDE plays a huge role in the experience. I noticed that when running Arquillian based test cases from within IntelliJ, the resources (project-defaults.yml, etc.) were not being picked up as part of the built archive. More details about the IDE behavior can be found in the issue.

Modifications
-------------
Updated the scenario generator to not filter out the resource folders.

Result
------
I can now run Arquillian based tests from within the IDE itself.

Ladicek commented on arquillian/adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/DefaultDeploymentScenarioGenerator.java:
```diff
@@ -92,20 +91,10 @@
 
         scanner.scanPackage();
 
-        Set<String> prefixes = codeSources.stream().map(e -> e.getLocation().toExternalForm()).collect(Collectors.toSet());
```
If we remove `prefixes`, we can also remove `codeSources` and everything related, as that's only used for this filtering.

roguexz replied:
@Ladicek I believe I will be able to remove the `try { codesources.add() }` block but keep the rest (as that is used for preparing the archive).

kenfinnigan commented:
I'm confused as to why the comment mentions 4.x when this PR is against 2.x

roguexz commented:
@kenfinnigan I figured the master line makes it back to 4.x as well. If that's not the case, I can reword the comment.

kenfinnigan commented:
@roguexz 4.x and master are very different, so separate PRs are required if you want something in both

roguexz commented:
@kenfinnigan Fixed the Git commit message (as well as the PR message).
@Ladicek Included the requested changes.

Commit:
[Windows] Potential fix for the SwarmProcess not terminating gracefully.

roguexz commented on tools/src/main/java/org/wildfly/swarm/tools/exec/SwarmProcess.java:
```diff
@@ -119,7 +119,9 @@ public int stop(long timeout, TimeUnit timeUnit) throws InterruptedException {
-            process.destroyForcibly();
+            // Attempt to terminate the process forcibly and also wait for the same amount of time as the base attempt.
+            // This should potentially take care of the issues with the process not terminating on Windows CI build.
+            process.destroyForcibly().waitFor(timeout, timeUnit);
```
Looking at the errors in the Windows build, looks like the process isn't terminating in 2 minutes, which in itself is weird and needs to be looked at. When invoking the "destroyForcibly()" method, we still need to check if the process is alive or not. Instead of adding a lot of logic, I am including an additional `waitFor` using the same arguments as the above invocation. Let's see if the Windows build goes through fine or not and how much longer does it take for completion.

roguexz commented:
Is there a problem with the Windows tests? I noticed the following,

The [MP config test](https://ci.wildfly-swarm.io/job/thorntail-pull-request-windows/io.thorntail.testsuite$thorntail-microprofile-tck-config/645/testReport/org.eclipse.microprofile.config.tck/ConfigProviderTest/testEnvironmentConfigSource/) failure seems to be due to the fact windows is case insensitive and we have two environment variables with [differing case](https://github.com/thorntail/thorntail/blob/master/testsuite/microprofile-tcks/config/pom.xml#L94). Not sure if this is intentional.

Please let me know your review comments regarding the PR.

Commit:
[Windows CI Build] Setup different profiles for the TCK test suite.

On Windows, we cannot have case sensitive environment variables. The TCK test suite has mixed case environment variables that cause the CI build (for windows) to fail. This fix creates two profiles - one for windows and one for other environments. In the Windows profile, we do not test for case sensitive environment variables.

roguexz commented:
Rebased and added a separate profile for executing the MP TCK tests for Windows. Hopefully, now the Windows CI build should proceed further.

roguexz commented:
@Ladicek I am not sure I can fix the rest of the windows failures 😄 .. For starters, I do not have access to a Windows VM .. Can you please review and let me know if I should make additional changes?

Ladicek commented:
I'm no expert on Windows nor MP Config TCK, but I definitely think that fixing MP Config TCK failures in this PR is _not_ expected. But thanks for doing that :-)

Ladicek commented on testsuite/microprofile-tcks/config/pom.xml:
```diff
@@ -79,31 +79,76 @@
+                     <environmentVariables>
+                        <my_int_property>45</my_int_property>
+                        <MY_BOOLEAN_PROPERTY>true</MY_BOOLEAN_PROPERTY>
+                        <MY_STRING_PROPERTY>woohoo</MY_STRING_PROPERTY>
```
I was wondering if just removing `my_string_property` on Windows would work, but the answer is yes. The MP Config TCK has a special case for Windows here: https://github.com/eclipse/microprofile-config/blob/828ee2fee4ccbef6701943bb96d7be187408fd63/tck/src/main/java/org/eclipse/microprofile/config/tck/CDIPropertyNameMatchingTest.java#L82-L88

Ladicek commented on testsuite/microprofile-tcks/config/pom.xml:
```diff
@@ -79,31 +79,76 @@
+                     <!-- https://issues.apache.org/jira/browse/SUREFIRE-569 -->
+                     <dependenciesToScan>
+                        <dependency>org.eclipse.microprofile.config:microprofile-config-tck</dependency>
+                     </dependenciesToScan>
```
Not to be changed in this PR, but I wonder if we need this. The [TCK documentation](https://github.com/eclipse/microprofile-config/blob/master/tck/running_the_tck.asciidoc) says that we either need `tck-suite.xml` and `<suiteXmlFile>`, _or_ we need `<dependenciesToScan>`. We have both. I wonder if that could possibly cause issues.

manovotn commented:
Win failures are unrelated to changes in this PR, going to merge.
